src/snapshot/src/lib.rs

```diff
@@ -128,25 +128,38 @@ impl Snapshot {
     }
 
     /// Attempts to load an existing snapshot and validate CRC.
-    pub fn load_with_crc64<T, O>(reader: &mut T, version_map: VersionMap) -> Result<O, Error>
+    pub fn load_with_crc64<T, O>(
+        reader: &mut T,
+        snapshot_len: usize,
+        version_map: VersionMap,
+    ) -> Result<O, Error>
     where
         T: Read,
         O: Versionize,
     {
         let mut crc_reader = CRC64Reader::new(reader);
-        let format_vm = Self::format_version_map();
-        let object: O = Snapshot::load(&mut crc_reader, version_map)?;
+
+        // Extract snapshot data without stored checksum, which is 8 bytes in size
+        let raw_snapshot_len = snapshot_len - std::mem::size_of::<u64>();
+        let mut snapshot = vec![0u8; raw_snapshot_len];
+        let bytes_read = crc_reader.read(&mut snapshot).unwrap_or(0);
```

Comment:
AFAIK `read()` does not guarantee reading all:

https://doc.rust-lang.org/std/io/trait.Read.html
> It is not an error if the returned value n is smaller than the buffer size, even when the reader is not at the end of the stream yet. This may happen for example because fewer bytes are actually available right now (e. g. being close to end-of-file) or because read() was interrupted by a signal.

Please use https://doc.rust-lang.org/std/io/trait.Read.html#method.read_exact

I'm also not sure if we need some sort of `BufferedReader` to avoid multiple underlying syscalls when using simple `Read`.

In a simple implementation I think the tradeoff between double buffering (using BufferedRead) and possible multiple syscalls would be more efficient to use double buffering...

@sandreim WDYT?

Reply:
Done